connectors/aws/aws-bedrock/src/main/java/io/camunda/connector/aws/bedrock/model/ConverseData.java

```diff
@@ -89,20 +90,24 @@ public final class ConverseData implements RequestData {
       binding = @TemplateProperty.PropertyBinding(name = "data.topP"))
   private Float topP = 0.9f;
 
+  @TemplateProperty(
+      label = "documents",
+      group = "converse",
+      id = "data.newDocuments",
+      feel = Property.FeelMode.required,
+      optional = true,
+      binding = @TemplateProperty.PropertyBinding(name = "data.newDocuments"))
+  private List<Document> newDocuments;
+
   @Override
-  public BedrockResponse execute(
+  public List<BedrockMessage> execute(
       BedrockRuntimeClient bedrockRuntimeClient, ObjectMapper mapperInstance) {
-    this.messages.add(new PreviousMessage(this.nextMessage, ConversationRole.USER.name()));
-    Message.Builder messageBuilder = Message.builder();
-    List<Message> messages =
-        this.messages.stream()
-            .map(
-                message ->
-                    messageBuilder
-                        .role(ConversationRole.valueOf(message.role()))
-                        .content(ContentBlock.fromText(message.message()))
-                        .build())
-            .toList();
+    var messageMapper = createMessageMapper();
+    var bedrockContentMapper = messageMapper.getBedrockContentMapper();
+    this.messagesHistory.add(prepareNewBedrockMessage(bedrockContentMapper));
```

Comment:
Having `prepareNewBedrockMessage` be part of MessageMapper would be clearer

Reply:
make sense, I will do ))

Reply:
done